Hardware/Sources/Hardware/CardReaderEvent.swift

```diff
@@ -0,0 +1,14 @@
+/// The possible events from a connected reader.
+public enum CardReaderEvent {
+    // A card was inserted.
+    case cardInserted
+
+    // A card was removed.
+    case cardRemoved
+
+    // Low battery warning.
+    case lowBattery
```

Comment:
Maybe add lowBatteryResolved. Maybe also add softwareUpdateNeeded and softwareUpToDate. OK to wait for next iteration for all of this.